A30-xds-v3.md

```diff
@@ -0,0 +1,255 @@
+binaries in lock step with their xDS server is if they are migrating
+their xDS server from version N-1 to version N+1 in a single step,
+without adding support for version N in between.  We do not recommend
+this kind of transition.
+
+### Details of the v2 to v3 Transition
+
+The xDS major version number appears in two main places in the API:
+
+- The RPC method name used to make the xDS request.  For v2, this is
+  `/envoy.service.discovery.v2.AggregatedDiscoveryService/StreamAggregatedResources`,
+  whereas for v3 it is
+  `/envoy.service.discovery.v3.AggregatedDiscoveryService/StreamAggregatedResources`.
+  We call this the *transport protocol version*.
+- The type of resources that we request via the transport protocol.  For
+  example, the type for a Listener resource in v2 is `envoy.api.v2.Listener`,
+  whereas in v3 the type is `envoy.config.listener.v3.Listener`.  We
+  call this the *resource version*.
+
+When gRPC uses v3 to talk to a server, the primary thing that it is
+choosing is the transport protocol version, not the resource version.
+
+Luckily, we can be more flexible with the resource version, because
+there's essentially only one case where we use a field in the v2 protos
+that was removed in the v3 protos.  (That case is the node `build_version`
+field, which we can handle as a special case.)  Therefore, we can
+simply change our code to parse all incoming messages as v3, regardless
+of whether they are actually v2 or v3 messages.  We will also need to
+accept both v2 and v3 type names interchangeably in all `type_url` fields.
+
+This means that the logic for determining the transport protocol version
+is completely independent of the logic for handling the resource version.
+gRPC will always request resources of the same version as the transport
+protocol it is using.  However, it will accept any supported version of
+resources in responses from the server, regardless of what version of
+the transport protocol we are using.
+
+In other words, if we're using the v2 transport protocol, we will
+request v2 resources, but it's fine for the server to send us either v2
+or v3 resources.  Similarly, if we're using the v3 transport protocol,
+we will request v3 resources, but it's fine for the server to send us
+either v2 or v3 resources.
+
+### Version of LRS Protocol
+
+For now, we will use the same server feature in the bootstrap file to
+indicate both the xDS and the LRS transport protocol version to use.
+
+Note that this means that there will not be a way to configure the client
+to use different transport versions for the two protocols.  In the future,
+if this becomes a problem, we can make use of `transport_api_version`
+field being added in https://github.com/envoyproxy/envoy/pull/11754 to
+specify the transport version of LRS.  However, since this field will exist
+only in xDS v3, this would still not provide a clean way to specify use
+of LRS v3 while using xDS v2.
```

Comment:
I think maybe we could add this to v2. The v2 freeze is there to encourage folks to move to v3, if we're adding a field to explicitly make this easier, then why not.

Reply:
Sounds good.  I've sent you envoyproxy/envoy#11824.